Objects/listobject.c

```diff
@@ -80,9 +122,39 @@ list_resize(PyListObject *self, Py_ssize_t newsize)
 
     if (newsize == 0)
         new_allocated = 0;
+
+#ifdef Py_GIL_DISABLED
+    _PyListArray *array = NULL;
+    if (new_allocated <= (size_t)PY_SSIZE_T_MAX / sizeof(PyObject *)) {
+        array = list_allocate_array(new_allocated);
+    }
+    else {
+        // integer overflow
+        array = NULL;
+    }
+    if (array == NULL) {
+        PyErr_NoMemory();
+        return -1;
+    }
+    PyObject **old_items = self->ob_item;
+    if (self->ob_item) {
+        if (allocated < new_allocated) {
+            memcpy(&array->ob_item, self->ob_item, allocated * sizeof(PyObject*));
+        }
+        else {
+            memcpy(&array->ob_item, self->ob_item, new_allocated * sizeof(PyObject*));
+        }
+    }
+     _Py_atomic_store_ptr_release(&self->ob_item, &array->ob_item);
```

Comment:
I thought that this atomic operation is needed to ensure memory ordering before we try to store array->ob_item into self->ob_item.

But please let me know if there is no possibility of a reordering issue for this operation.

Reply:
Yeah, that's right